src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -193,7 +193,7 @@ protected override ImmutableArray<PendingBranch> Scan(ref bool badRegion)
             DiagnosticBag diagnostics,
             Action<BoundExpression, TypeSymbolWithAnnotations> callbackOpt = null)
         {
-            if (method.IsImplicitlyDeclared)
+            if (method.IsImplicitlyDeclared && (!method.IsImplicitConstructor || method.ContainingType.IsImplicitlyDeclared))
```

Comment:
>method.ContainingType.IsImplicitlyDeclared [](start = 81, length = 42)

Which test is demonstrating this part of the check?

Reply:
This check prevents analyzing the implicit constructor of the (implicit) script class in particular. Without this check, several scripting tests fail within NullableWalker.

---
In reply to: [253556084](https://github.com/dotnet/roslyn/pull/33093#discussion_r253556084) [](ancestors = 253556084)

Reply:
In the script case, the constructor and containing type are both implicit and the `if` condition should be `true`.

---
In reply to: [253564880](https://github.com/dotnet/roslyn/pull/33093#discussion_r253564880) [](ancestors = 253564880,253562102,253556084)